commands/create_user

```diff
@@ -52,24 +51,98 @@ while [ $# -gt 0 ]; do
+        # Check if the role argument is valid
+        valid_role=false
+        for valid_role_name in "${valid_roles[@]}"; do
+          if [[ "$valid_role_name" == "$role_arg" ]]; then
+            role="$role_arg"
+            valid_role=true
+            break
+          fi
+        done
+
+        if ! $valid_role; then
+          invalid_role_message="Error: invalid role '$role_arg'. Role must be one of: ${valid_roles[0]}"
+          for (( i=1; i<${#valid_roles[@]}; i++ )); do
+            invalid_role_message+=", ${valid_roles[$i]}"
+          done
+          echo "$invalid_role_message" >&2
+          exit 1
+        fi
+      else
+        echo "Option $1 requires an argument." >&2
+        exit 1
+      fi
+      ;;
     *)
       REST_ARGS+=("$1")
       ;;
   esac
   shift
 done
+
+echo "URL       :  $C8Y_HOST"
+echo "Tenant ID :  $C8Y_TENANT"
+echo "User      :  $C8Y_USER"
+
+# Check that the required arguments are present
+missing_args=()
+if [[ -z $username ]]; then
+  missing_args+=(username)
+fi
+
+if [[ -z $password ]]; then
+  missing_args+=(password)
+fi
+
+if [[ -z $email ]]; then
+  missing_args+=(email)
+fi
+
+if [[ -z $role ]]; then
+  missing_args+=(role)
+fi
+
+if [[ ${#missing_args[@]} -gt 0 ]]; then
+  message="Error: Missing required arguments: ${missing_args[0]}"
+  for (( i=1; i<${#missing_args[@]}; i++ )); do
+    message+=", ${missing_args[$i]}"
+  done
+  echo "$message"
+  exit 1
+fi
+
 set -- "${REST_ARGS[@]}"
 
 oeeadminUser=$(c8y users list --includeAll --username "$username" -o csv --select "id" -n)
 
 if [ -z "$oeeadminUser" ]; then
   echo "$username user was not detected, creating $username user."
 
-  echo "Create user $username and add to OEE Admin, OEE Reader, business groups"
-  c8y users create -f -n --userName "$username" --password "$password" --email "$email" | \
-    c8y userreferences addUserToGroup -f --group "business"   | \
-    c8y userreferences addUserToGroup -f --group "OEE Admin"  | \
-    c8y userreferences addUserToGroup -f --group "OEE Reader"
+  if [ "$role" == "admin" ]; then
+    response=$(c8y users create -f -n --userName "$username" --password "$password" --email "$email" | \
+                c8y userreferences addUserToGroup -f --group "business"   | \
+                c8y userreferences addUserToGroup -f --group "OEE Admin" 2>&1)
+  elif [ "$role" == "reader" ]; then
+    response=$(c8y users create -f -n --userName "$username" --password "$password" --email "$email" | \
+                c8y userreferences addUserToGroup -f --group "business"   | \
+                c8y userreferences addUserToGroup -f --group "OEE Reader" 2>&1)
+  elif [ "$role" == "access" ]; then
+    response=$(c8y users create -f -n --userName "$username" --password "$password" --email "$email" | \
+                c8y userreferences addUserToGroup -f --group "business" 2>&1)
+  fi
+
+  output=$(c8y users getUserByName --name "$username" 2>&1)
+  if [[ $output == *"serverError"* ]]; then
```

Comment:
couldnt we check by using the response code? should be 404 in case the user was not created?

Reply:
Yes, we can check 404 error but in case that the error relates to other error then we need to add another condition check which increase the complexity of the script. I just add the common serverError check and echo the response so that user can know what goes wrong